cli/npm/dsl.d.ts

```diff
@@ -344,6 +345,20 @@ declare const token: {
   immediate(rule: RuleOrLiteral): ImmediateTokenRule;
 };
 
+/**
+ * Creates an eof rule, matching the end of the input.
+ * This is useful in languages where the top level grammar is a repeated
+ * sequence of statements followed by a terminator. One might use newlines
+ * for a terminator, but the end of the file also should match. Previously,
+ * '\0' was used to match eof, but this can match any NUL byte in the input,
```

Comment:
I'd remove the `Previously, '\0' was used` sentence from the docs. I'm assuming you mean some specific grammar used `\0` to detect EOF? Either way, I don't think the docs need to talk about some thing that previously existed.